src/openforms/submissions/tests/factories.py

```diff
@@ -68,6 +68,7 @@ def from_components(
         cls,
         components_list: List[dict],
         submitted_data: dict = None,
+        with_report=True,
```

Comment:
would be better suited as a trait:

```py
with_report = factory.Trait(
    report=factory.SubFactory(SubmissionReportFactory, submission=factory.SelfAttribute(".."))
)
```

not sure if this exact invocation of `SelfAttribute` works, but it should be something like that. See docs: https://factoryboy.readthedocs.io/en/stable/recipes.html#copying-fields-to-a-subfactory